pallets/liquid-staking/src/lib.rs

```diff
@@ -885,20 +913,33 @@ pub mod pallet {
                 .unwrap_or_else(Zero::zero)
         }
 
-        #[allow(dead_code)]
         fn bonded_of(index: DerivativeIndex) -> BalanceOf<T> {
-            StakingLedgers::<T>::get(&index).map_or(Zero::zero(), |ledger| ledger.active)
+            Self::staking_ledger(&index).map_or(Zero::zero(), |ledger| ledger.active)
         }
 
         fn unbonding_of(index: DerivativeIndex) -> BalanceOf<T> {
-            StakingLedgers::<T>::get(&index).map_or(Zero::zero(), |ledger| {
+            Self::staking_ledger(&index).map_or(Zero::zero(), |ledger| {
                 ledger.total.saturating_sub(ledger.active)
             })
         }
 
-        fn has_unbonded(index: DerivativeIndex) -> bool {
-            StakingLedgers::<T>::get(&index)
-                .map_or(false, |ledger| ledger.has_unbonded(Self::current_era()))
+        fn unbonded_of(index: DerivativeIndex) -> BalanceOf<T> {
+            let current_era = Self::current_era();
+            Self::staking_ledger(&index).map_or(Zero::zero(), |ledger| {
+                ledger.unlocking.iter().fold(Zero::zero(), |acc, chunk| {
+                    if chunk.era <= current_era {
+                        acc.saturating_add(chunk.value)
+                    } else {
+                        acc
+                    }
+                })
+            })
+        }
```

Comment:
The maximum number of iterations is 32 here. and it is called in `do_multi_withdraw_unbonded`, suppose there are 8 accounts, the  number of iterations could be 32 * 8, and `do_multi_withdraw_unbonded` is in the `on_initial` function.